src/corehost/common/pal.h

```diff
@@ -54,12 +54,15 @@
 #if defined(_WIN32)
 #define LIB_PREFIX
 #define MAKE_LIBNAME(NAME) (_X(NAME) _X(".dll"))
+#define BASE_HOST_RID _X("win")
```

Comment:
2 thoughts:

1. No matter whether OSX 10.13 is compatible with its predecessors or not, in this plan when we officially move to support OSX 10.13, we would move the base fallback RID to be osx.10.13.
1. In the case where OSX 10.13 is incompatible with its predecessors and we still wanted to support osx.10.12 and earlier, we would need to build for OSX twice - osx.10.12 and osx.10.13 - just like we do for each major version of `ubuntu` today - ubuntu14.04 and ubuntu.16.04.

This fallback plan is not un-like what we did in `1.0` where the `TARGET_RUNTIME_ID` is supplied at build time.  The differences from `1.0` are:

1. For all platforms, we first try the actual machine's RID, if that works - great use it.  This allows us to update the `runtime.json` file, and the host automatically picks it up and works with no code changes. This change from `1.0` gives us a lot of future proofing.
2. When the machine's RID doesn't work:
  - When running on Linux, instead of using the RID of the machine we built on (since we have absolutely no idea of compatibility between this unknown Linux distro and the distro we built on), we fall back to `linux` RID, because that is the common ground for all Linux distros.
  - When on OSX and Windows, if we are running on an unknown RID, there are really only 2 possible situations:  we are on a higher version or a lower version.  If we are on a lower version, we never said we support it, and chances are we probably didn't even get this far.  But on the higher version case, I would expect things to work, barring any OS incompatibilities.  So to make it even possible for the app to work on newer versions, we start with the highest OS version we knew at build time.

Given this plan, we won't HAVE to update the host code when a new OSX version comes out.  We just need to update the `runtime.json` file, after we've done testing, and the new RID will start to work in the fallback graph.  But until that time, which takes a while, we should at least give the apps a chance to run on the new OSX.  That way we can evaluate/test on the new OSX version without making any changes at all.